client/packages/core/src/Reactor.js

```diff
@@ -292,25 +294,37 @@ export default class Reactor {
 
   /**
    * @param {'enqueued' | 'pending' | 'synced' | 'timeout' |  'error' } status
-   * @param string eventId
-   * @param {{message?: string, hint?: string, error?: Error}} [errDetails]
+   * @param {string} eventId
+   * @param {{message?: string, type?: string, status?: number, hint?: unknown}} [errorMsg]
    */
-  _finishTransaction(status, eventId, errDetails) {
+  _finishTransaction(status, eventId, errorMsg) {
     const dfd = this.mutationDeferredStore.get(eventId);
     this.mutationDeferredStore.delete(eventId);
     const ok = status !== 'error' && status !== 'timeout';
 
     if (!dfd && !ok) {
       // console.erroring here, as there are no listeners to let know
-      console.error('Mutation failed', { status, eventId, ...errDetails });
+      console.error('Mutation failed', { status, eventId, ...errorMsg });
     }
     if (!dfd) {
       return;
     }
     if (ok) {
       dfd.resolve({ status, eventId });
     } else {
-      dfd.reject({ status, eventId, ...errDetails });
+      // Check if error comes from server or client
```

Comment:
My thinking is, we should return the same `error` in both cases. 

In both cases imo we should throw dfd.reject(new InstantError(...)) 

Separately, we can make InstantAPIError extend InstantError. 

cc @dwwoelfel @nezaj if you have other thoughts

Reply:
I think it's a useful distinction to tell whether the error is from the server or the client and since InstantAPIError extends InstantError, checking if its InstantError will work 100% of the time